autovacuum_mail_message/models/message_vacuum_rule.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018 Akretion
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from openerp import api, fields, models, exceptions, _
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
+from datetime import date, timedelta
+
+
+class MessageVacuumRule(models.Model):
+    _name = "message.vacuum.rule"
+    _description = "Rules Used to delete message historic"
+
+    name = fields.Char(required=True)
+    company_id = fields.Many2one(
+        'res.company', string="Company",
+        default=lambda self: self.env['res.company']._company_default_get(
+            'message.vacuum.rule'))
+    message_subtype_ids = fields.Many2many(
+        'mail.message.subtype', string="Subtypes",
+        help="Message subtypes concerned by the rule. If left empty, the "
+             "system won't take the subtype into account to find the "
+             "messages to delete")
+    empty_subtype = fields.Boolean(
+        help="Take also into account messages with no subtypes")
+    model_ids = fields.Many2many(
+        'ir.model', string="Models",
+        help="Models concerned by the rule. If left empty, it will take all "
+             "models into account")
+    message_type = fields.Selection([
+        ('email', 'Email'),
+        ('comment', 'Comment'),
+        ('notification', 'System notification')], required=True)
```

Comment:
What if I want to erase all those types ? In this case it shouldn't required to delete all types ?

Reply:
To delete all those types, you can create 3 rules, one for each type.
I think it is not a too heavy configuration. The other solution would be to add a type 'all', maybe it is simpler you are right

Reply:
I have added the possibility to choose "all" within the  types.
I also fixed the pylint errors.